glslc/src/file_compiler.cc

```diff
@@ -77,6 +89,28 @@ bool FileCompiler::CompileShaderFile(const std::string& input_file,
                 << output_file_name_ << "'" << std::endl;
     }
   }
+
+  // Handle the error message for failing to deduce the shader kind.
+  if (result.GetCompilationStatus() ==
+      shaderc_compilation_status_invalid_stage) {
```

Comment:
In this situation, should we depend on `output_stream->write` succeeding on line 79?  If not, this check should happen before (and instead of) outputting.